HowToRPC.md

```diff
@@ -0,0 +1,64 @@
+
+- **Go** [github.com/golang/protobuf](https://github.com/golang/protobuf)
+- **Swift** [github.com/apple/swift-protobuf](https://github.com/apple/swift-protobuf)
+- **Rust** [github.com/stepancheg/rust-protobuf](https://github.com/stepancheg/rust-protobuf)
+
+### A way to make API requests
+[gRPC](https://grpc.io/) is the recommended way to call Google RPC APIs. gRPC support is typically provided by additional `protoc` plugins that generate code for API clients and servers. This code uses lower-level primitives that send messages using gRPC’s HTTP/2-based messaging system, which supports request multiplexing, streaming APIs, and advanced flow control. To learn more about working with gRPC, visit [grpc.io/docs](https://grpc.io/docs).
+
+If gRPC support is unavailable, Google APIs can also be called using HTTP/1 using the protocol described in the next section.
+
+### Authentication
+To use Google APIs, a client needs to use API keys for unauthenticated requests or OAuth tokens for authenticated requests. For more information, see [the Google Cloud Authentication Overview](https://cloud.google.com/docs/authentication/).
+
+API keys can be obtained from the [Google Cloud Console > Credentials](http://console.cloud.google.com/apis/credentials) page. OAuth tokens can be obtained by [OAuth 2](https://oauth.net/2/) clients and libraries. For a sample command-line client, see the [oauth2l](https://github.com/google/oauth2l) on GitHub.
+
+## The HTTP RPC Protocol
+Most Google APIs support a simple RPC protocol using Protocol Buffers (protobuf) over HTTP. It allows clients to call Google APIs directly, often using standard library functions.
+
+This protocol uses fixed URLs to specify the RPC endpoints, and passes request/response messages as HTTP request/response body using HTTP POST. It uses normal HTTP headers to pass the RPC metadata, such as System Parameters.
+
+### URL
+RPC URLs have the following format:
+
+```
+URL ::= BaseUrl "/" Service "/" Method
+```
+- **BaseUrl.** This is the base URL published by service owners, either via documentation or service discovery. For most Google APIs, the BaseUrl looks like "https://language.googleapis.com/$rpc". The base address can be found in the API reference documentation where it is identified as the “Service name”. For this example, “language.googleapis.com” is found in the Google Cloud Natural Language API Reference. 
+- **Service.** This is the fully qualified protobuf `service` name, such as "google.cloud.language.v1.LanguageService". In this case, “google.cloud.language.v1” is the package name in google/cloud/language/v1/language_service.proto and “LanguageService” is the name of the service section found on this line.
+- **Method.** This is the protobuf `rpc` name, such as "AnnotateText".
+
+### Requests
+RPC request messages are serialized and sent as the HTTP request body. For the server to handle the request properly, the client must set several HTTP request headers:
```

Comment:
I strongly prefer 80-column or 100-column per line.

Reply:
Agreed, and 80, please.

Reply:
That's pretty horrible for editing. Do you use a text editor that automatically manages line breaks?

Reply:
I've auto-wrapped all Markdown using [pretty.sh](https://github.com/timburks/googleapis.github.io/blob/master/pretty.sh) (in the repo).